### PR TITLE
Start the slicer custom application

### LINKED ISSUE #1: initialize the application with SlicerOpenLIFU bundled into it
Once https://github.com/OpenwaterHealth/SlicerOpenLIFU/issues/1 is completed, we can create a custom application using Kitware's [SlicerCustomApplicationTemplate](https://github.com/KitwareMedical/SlicerCustomAppTemplate). We will configure this to bundle the SlicerOpenLIFU extension and use SlicerOpenLIFU's placeholder module as its "Home" module

### PR DESCRIPTION
**Pull request by ebrahimebrahim**
Closes #1

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+message(STATUS "Remote - ${extension_name} [OK]")
+list(APPEND Slicer_EXTENSION_SOURCE_DIRS ${${extension_name}_SOURCE_DIR}/Modules/Scripted/SlicerCustomAppUtilities)
+
+# Bundle the SlicerOpenLIFU extension
+set(extension_name "SlicerOpenLIFU")
+set(${extension_name}_SOURCE_DIR "${CMAKE_BINARY_DIR}/${extension_name}")
+FetchContent_Populate(${extension_name}
+ SOURCE_DIR     ${${extension_name}_SOURCE_DIR}
+ GIT_REPOSITORY ${EP_GIT_PROTOCOL}://github.com/OpenwaterHealth/SlicerOpenLIFU.git
+ GIT_TAG        0ea13d4fd39937059085f51fd14f60e6eca81397
```
Comment (ebrahimebrahim) [inline CMakeLists.txt:167]: This tag may have to change depending on the fate of https://github.com/OpenwaterHealth/SlicerOpenLIFU/pull/2

File: Modules/Scripted/Home/Home.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        self.parent.contributors = ["Sam Horvath (Kitware Inc.)", "Jean-Christophe Fillion-Robin (Kitware Inc.)"]
+        self.parent.helpText = """This module orchestrates and styles the overall application workflow."""
+        self.parent.helpText += self.getDefaultModuleDocumentationLink()
+        self.parent.acknowledgementText = """..."""  # replace with organization, grant and thanks.
+
+        # Create the Home module's widget representation just to ensure that HomeWidget.setup
+        # gets called and the styling and UI modifications are run.
+        def touch_widget_representation():
+                slicer.modules.home.widgetRepresentation()
+        slicer.app.connect("startupCompleted()", touch_widget_representation)
```
Comment (ebrahimebrahim) [inline Modules/Scripted/Home/Home.py:40]: (@sjh26 please see this especially -- this is how I ended up getting your suggestion to work.)

Comment (sjh26) [reply Modules/Scripted/Home/Home.py:40]: lgtm, seems to work fine, and is a nice way to make sure the entire widget/logic is set up

**Review by sjh26 — APPROVED** [review]

**Comment by sjh26** [comment]
Once this https://github.com/Slicer/Slicer/pull/7833 slicer PR is approved, we should update the Slicer version (we can do that as a separate issue).  That bug affects the Windows build.

**Pull request merged**